assets/src/edit-story/app/rightClickMenu/provider.js

```diff
@@ -99,11 +126,66 @@ function RightClickMenuProvider({ children }) {
     ev.stopPropagation();
   }, []);
 
+  /**
+   * Copy the element and styles to the clipboard.
+   */
+  const handleCopyElement = useCallback(async () => {
+    if (!selectedElements.length) {
+      return;
+    }
+
+    const { htmlContent, serializedPayload } = serializeTextAndHTMLData(
+      currentPage,
+      selectedElements,
+      selectedElementAnimations
+    );
+
+    try {
+      await navigator.clipboard.writeText(
+        `<!-- ${serializedPayload} -->${htmlContent}`
+      );
+    } catch (error) {
```

Comment:
‼️ This is important! This is the new [cliboard api])(https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API). `writeText` and `readText` are supported on all browsers that we support but we're only encoding the html content atm.

The old copy/paste logic adds html and text components when copy and pasting. So this doesn't support as much as the old logic. The old logic still exists here: https://github.com/google/web-stories-wp/blob/main/assets/src/edit-story/utils/copyPaste.js